test/chemistry/test_adapt_vqe.py

```diff
@@ -0,0 +1,139 @@
+from qiskit.aqua.components.optimizers import L_BFGS_B
+from qiskit.chemistry.ground_state_calculation import AdaptVQE, VQEUCCSDFactory
+from qiskit.chemistry.qubit_transformations import FermionicTransformation
+
+
+class TestAdaptVQE(QiskitChemistryTestCase):
+    """ Test Adaptive VQE Ground State Calculation """
+
+    def setUp(self):
+        super().setUp()
+
+        try:
+            self.driver = PySCFDriver(atom='H .0 .0 .0; H .0 .0 0.735',
+                                      unit=UnitsType.ANGSTROM,
+                                      basis='sto3g')
+        except QiskitChemistryError:
+            self.skipTest('PYSCF driver does not appear to be installed')
+            return
+
+        self.expected = -1.85727503
+
+        self.transformation = FermionicTransformation()
+
+    def test_default(self):
+        """ Default execution """
+        solver = VQEUCCSDFactory(QuantumInstance(BasicAer.get_backend('statevector_simulator')))
+        calc = AdaptVQE(self.transformation, solver)
+        res = calc.compute_groundstate(self.driver)
+        self.assertAlmostEqual(res.electronic_energy, self.expected, places=6)
+
+    def test_custom_minimum_eigensolver(self):
+        """ Test custom MES """
+        solver = VQEUCCSDFactory(QuantumInstance(BasicAer.get_backend('statevector_simulator')))
+
+        def get_custom_solver(self, transformation):
+            num_orbitals = transformation._molecule_info['num_orbitals']
+            num_particles = transformation._molecule_info['num_particles']
+            qubit_mapping = transformation._qubit_mapping
+            two_qubit_reduction = transformation._molecule_info['two_qubit_reduction']
+            z2_symmetries = transformation._molecule_info['z2_symmetries']
+            initial_state = HartreeFock(num_orbitals, num_particles, qubit_mapping,
+                                        two_qubit_reduction, z2_symmetries.sq_list)
+            var_form = UCCSD(num_orbitals=num_orbitals,
+                             num_particles=num_particles,
+                             initial_state=initial_state,
+                             qubit_mapping=qubit_mapping,
+                             two_qubit_reduction=two_qubit_reduction,
+                             z2_symmetries=z2_symmetries)
+            vqe = VQE(var_form=var_form, quantum_instance=self._quantum_instance,
+                      optimizer=L_BFGS_B())
+            return vqe
+
+        # pylint: disable=no-value-for-parameter
+        solver.get_solver = get_custom_solver.__get__(solver, VQEUCCSDFactory)
+
+        calc = AdaptVQE(self.transformation, solver)
+        res = calc.compute_groundstate(self.driver)
+        self.assertAlmostEqual(res.electronic_energy, self.expected, places=6)
+
+    def test_custom_excitation_pool(self):
+        """ Test custom excitation pool """
+        # TODO rewrite this unittest once we have reworked how AdaptVQE will handle the solver
```

Comment:
I think this TODO is obsolete as of now. @stefan-woerner could you please comment?